### DISCUSSION TITLE
Option to restrict TX on PMR446 frequencies

### DISCUSSION THREAD

**Discussion by coilert** [egzumer/uv-k5-firmware-custom#375] [Ideas]
Just joining the radio club using a Quangsheng UV-5R+. As I do not have a radio licence I wondered if it could be an option to restrict TX on the PMR446 frequencies? Also the allowed TX power for unlicenced PMR use is 500mW (EU), not sure if that would be possible to change though...

Edit: Already found the discussion topic on the power limits so you may disregard my last question.

**Comment by Arikshten** [comment]
### Restrict TX on PMR446
When programming a radio using CHIRP If you set duplex to ‘off’ instead of ‘none’ that will prevent TX on that memory slot. I use this myself to prevent TX on specific frequencies in memory and this could perhaps be an option for you. 

As a side note: the Quangsheng is obviously not legal to use on PMR446 due in part to it's removable antenna as well as the power output being above .5W. Even if we are able to limit the power output it is still not within the restrictions. 

If you are unlicensed then surely you would want to simply disable TX everywhere..  Which can be done from the option 61 (currently) 
"F Lock - sets the TX frequency band plan", in the hidden menu (https://github.com/egzumer/uv-k5-firmware-custom/wiki/Menu#hidden-menu).


### Restrict TX to PMR446
Although -- Perhaps I have misunderstood and you may be asking to "restrict TX **to** the PMR446" only.    
In which case you can always simply use CHIRP to only program in memory channels within PMR446. 
Otherwise we would have to ask for an enhancement to the menu option "F Lock - sets the TX frequency band plan" to allow TX in PMR446 only. 
I am not sure how popular or useful such an option would be as one can always simply chose to not broadcast.


### Perhaps already available
Aside from that you can always customise the firmware yourself to do that very easily. 
UVMOD has an option for "Custom TX Range" ..  
Have a look at  https://whosmatt.github.io/uvmod/index.html
If you enable the hidden mods you will see an option for the "Custom TX Range"


I hope that is of help ..

**Reply by ErikS-web** [reply]
This does require an option "**Only MR**" or "**NO-VFO**". 

With the intention that the VFO cannot be switched on.
But what about the frequentiescanning-mode?

**Reply by egzumer** [reply]
You can disable frequency mode with the chirp

**Reply by coilert** [reply]
Thanks for your quick replies! I was indeed referring to adding an extra F-Lock option to avoid any accidental transmissions outside the EU PMR band while playing with the radio. Of course I do realize that this does not make the radio legal because of the reasons mentioned by @Arikshten but it may still avoid getting into trouble. 

My idea is to add an additional F-lock item and check if the frequency is set to one of the 16 allowed PMR446 frequencies like this:

frequencies.c
```c
		case F_LOCK_PMR446:
			if (Frequency >= 44600625 && Frequency < 44620000 && ((Frequency-625) % 1250 == 0))
				return 0;
			break;
```

**Comment by ErikS-web** [comment]
Indeed already possible.

Use Chirp with EGZUMER-driver.
Make your Memory Channel list 
and modify this Advanged setting

![Knipsel](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/71377d01-dd6f-465e-b490-95bc0e100639)

**Reply by coilert** [reply]
Thanks again for your help! As an aviation enthousiast however, one of the key reasons of my interest in this project is the improvement in reception of aviation communication. For this, having the frequency mode enabled is of course very useful, especially when going to other places with different local aviation frequencies. 

So my use-case is mainly RX but it would be great to have TX enabled for the PMR band to safely experiment as well.
